### DISCUSSION TITLE
How can I get Deepgram to send Authorization headers to Twilio to grab call recordings?

### DISCUSSION THREAD

**Discussion by daveidivide** [deepgram/community#484] [General help] [answered]
Hi, I am trying to get Deepgram to get MP3s from my Twillio account and transcribe them.

I am sending a POST request to `https://api.deepgram.com/v1/listen?filler_words=false&summarize=v2` with the following body:

`{
  "url": "https://api.twilio.com/2010-04-01/Accounts/{my twilio account id here}/Recordings/{my twilio recording id here}.mp3",
  "headers": {
      "Authorization": "Basic {My Twilio auth token here}"
  }
}`

The response I get from deepgram is:
`{
    "err_code": "Bad Request",
    "err_msg": "The remote server hosting the media returned a client error: 401 Unauthorized",
    "request_id": "43c84002-75e7-447d-ba48-40f0d12f737a"
}`

When I make the request myself to Twilio it works fine (as it is sending the Authorization token along with the request).

How can I get deepgram to add in this Authorization token in the headers?

Essentially, is it possible to get deepgram to access an mp3 that requires auth?

Many thanks for any help :)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment, marked as answer]
Hey @daveidivide, have you taken a look at our Twilio guide? https://developers.deepgram.com/docs/on-premise-twilio-integration It has code examples and a workflow for integrating Deepgram with Twilio.

**Reply by jjmaldonis** [reply]
Hey @daveidivide did you get this working?

Here is another blog post with more up to date info: https://deepgram.com/learn/deepgram-twilio-streaming

**Reply by daveidivide** [reply]
I ended up realising that in the future I will very likely need the call recording stored on my own services, so I rewrote it so that I download the mp3, store in my S3 bucket and then point deepgram at my own endpoint.

I will check out the blog post though.

Thanks!

**Reply by jjmaldonis** [reply]
Very good to know, thanks for the update!
